qt/scientific_interfaces/Inelastic/QENSFitting/FittingModel.cpp

```diff
@@ -597,7 +601,11 @@ std::string FittingModel::singleFitOutputName(std::string workspaceName, Workspa
   return inputWorkspace + "_" + m_fitType + "_" + m_fitString + "_" + spectra + "_Results";
 }
 
-std::string FittingModel::getOutputBasename() const { return cutLastOf(sequentialFitOutputName(), "_Results"); }
+std::optional<std::string> FittingModel::getOutputBasename() const {
+  if (auto const outputName = sequentialFitOutputName())
```

Comment:
Sorry, I'm not sure I understand -  I think `sequentialFitOutputName` already returns a std::optional?

Reply:
I think `sequentialFitOutputName` is called from elsewhere in the FittingModel

Reply:
What I meant is that instead of make both `sequentialFitOutputName` and `simultaneousFitOutputName` as std::optional, you could return the optional on `getOutputBaseName`, and just the string (or empty string) on the other functions. 
The other place in which both functions are called is from the `createSimultaneous/SequentialFit...` functions. But I think in the last commit you have fix that in the case it returns a nullopt. So, it is fine.

Reply:
Ah I see, I think I prefer it returns std::optional rather than an empty string because it makes it more explicit that the function could return an empty result. That way its easier to spot assumptions like the one you spotted in those `createSimultaneous/SequentialFit...` functions. Are you happy if I keep this the same?

Reply:
Yes, makes sense. Thank you.